### PR TITLE
feat: Fetch multiple pages of NFT offers, add key and blockchain

### PR DESCRIPTION
**Pull request by hanjano**

### CODE REVIEW

File: blockapi/v2/api/nft/opensea.py [hunk 1]
```diff
@@ -113,50 +114,37 @@ def parse_nfts(self, fetch_result: FetchResult) -> ParseResult:
...
-        return ParseResult(
-            data=list(
-                self._yield_offers(
-                    NftOfferDirection.OFFER,
-                    fetch_result.extra,
-                    fetch_result.data.get('offers'),
-                )
+        parsed = list(
+            self._yield_parsed_offers(
+                NftOfferDirection.OFFER, 'offers', fetch_result.data, fetch_result.extra
```
Comment (ladovina) [inline blockapi/v2/api/nft/opensea.py:126]: what about joining the first two params into one, `offer` vs `offers`?

File: blockapi/v2/api/nft/opensea.py [hunk 2]
```diff
@@ -190,42 +178,83 @@ def parse_collection(self, fetch_result: FetchResult) -> ParseResult:
...
+            fetched, next_cursor = fetch_method(key, cursor)
+            yield fetched
+
+            if not next_cursor:
                 break
 
+            cursor = next_cursor
+            if cursor in cursors:
+                logger.warning(
+                    f'Detected duplicate cursor {cursor} while fetching NFTs for {address}'
```
Comment (ladovina) [inline blockapi/v2/api/nft/opensea.py:220]: is variable `address` undefined?

File: blockapi/v2/api/nft/opensea.py [hunk 3]
```diff
@@ -331,16 +360,18 @@ def _parse_collection_stats(
 
             return st
 
-    def _yield_offers(
-        self, direction: NftOfferDirection, extra: dict, items: list
+    def _yield_parsed_offers(
+        self, direction: NftOfferDirection, key: str, results: dict, extra: dict
```
Comment (ladovina) [inline blockapi/v2/api/nft/opensea.py:364]: `result` can be a list or a dict, right?

File: blockapi/v2/api/nft/opensea.py (hunk 2 above)
Comment (hanjano) [reply blockapi/v2/api/nft/opensea.py]: this should be fixed

File: blockapi/v2/api/nft/opensea.py (hunk 3 above)
Comment (hanjano) [reply blockapi/v2/api/nft/opensea.py]: it is actually always a list

**Review by ladovina — APPROVED** [review]

**Pull request merged**